src/client/js/dialog.js

```diff
@@ -0,0 +1,75 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const main = document.querySelector('body > main')
+const observer = new MutationObserver(handleMutation)
+const triggerLinks = main.querySelectorAll('a[href*="dialog/"]')
+let dialogEl
+
+function init (links) {
+  if (!dialogEl) {
+    dialogEl = document.createElement('dialog')
+    document.body.append(dialogEl)
+  }
+
+  links.forEach(link => link.addEventListener('click', handleEvent))
+}
+
+function handleMutation (mutationList) {
+  for (const mutation of mutationList) {
+    if (!mutation.addedNodes.length) continue // ignore removed-node mutations
+    const triggerLink = mutation.target.querySelector('a[href*="dialog/"]')
+    if (triggerLink) init([triggerLink])
+  }
+}
+
+function handleEvent (e) {
+  switch (true) {
+    case e.target.matches('a[href*="dialog/"]'):
+      e.preventDefault()
+      openDialog(e.target.href)
+      break
+    case e.target.matches('dialog button.close'):
+      dialogEl.close()
+      break
+  }
+}
+
+async function openDialog (path) {
+  const partialName = path.substring(path.lastIndexOf('/') + 1)
+
+  dialogEl.showModal() // provide immediate UI response by showing ::backdrop regardless of content load
+  dialogEl.setAttribute('data-partial', partialName) // allow selector access, e.g. dialog[data-partial='add-email']
+  dialogEl.addEventListener('click', handleEvent)
+  dialogEl.addEventListener('close', resetDialog)
```

Comment:
Suggestion: I’d expect clicking the backdrop of the dialog to dismiss it a well.

Reply:
I've always been a click-backdrop-to-dismiss person too.  The research I did left me on the fence. Using the native dialog element makes it non-trivial to add (with `::backdrop`).  In fact, I think you were the one who pointed out a bug where clicking on the dialog parent incorrectly closed it!  

Rather than throw more code at the feature I just removed it.  I support adding click-backdrop-to-dismiss functionality under separate ticket. 👍